test/yapi_network_test.rb

```diff
@@ -0,0 +1,374 @@
+        allow(Yast::Routing).to receive(:Write)
+      end
+
+      subject { Yast::YaPI::NETWORK.Write("route" => route) }
+
+      let(:success) { { "error" => "", "exit" => "0" } }
+
+      context "with no gateway in the default route" do
+        let(:route) { { "default" => nil } }
+
+        it "returns success" do
+          expect(subject).to eq success
+        end
+
+        it "empties the routes" do
+          expect(Yast::Routing).to receive(:Routes=).with []
+          subject
+        end
+      end
+
+      context "with empty gateway in the default  route" do
+        let(:route) { { "default" => { "via" => "" } } }
+
+        it "returns success" do
+          expect(subject).to eq success
+        end
+
+        it "empties the routes" do
+          expect(Yast::Routing).to receive(:Routes=).with []
+          subject
+        end
+      end
+
+      context "with a gateway in the default route" do
+        let(:route) { { "default" => { "via" => "TheIP" } } }
+        before do
+          expect(Yast::IP).to receive(:Check4).with("TheIP").and_return valid
+        end
+
+        context "if it's a valid IP4" do
+          let(:valid) { true }
+
+          it "returns success" do
+            expect(subject).to eq success
+          end
+
+          it "correctly modifies the routes" do
+            expect(Yast::Routing).to receive(:Routes=).with(
+              [
+                {
+                  "destination" => "default",
+                  "gateway"     => "TheIP",
+                  "netmask"     => "-",
+                  "device"      => "-"
+                }
+              ]
+            )
+            subject
+          end
+        end
+
+        context "if it's not a valid IP4" do
+          let(:valid) { false }
+
+          it "returns failure" do
+            res = subject
+            expect(res["exit"]).to eq "-1"
+            expect(res["error"]).to_not be_empty
+          end
+
+          it "doesnt't modify the routes" do
+            expect(Yast::Routing).to_not receive(:Routes=)
+            subject
+          end
+        end
+      end
+    end
+
+    context "with interfaces" do
+
+      subject { Yast::YaPI::NETWORK.Write("interface" => interface) }
+
+      before do
+        interface.keys.map do |k|
+          stub_clean_cache(k)
+        end
+      end
+
+      context("setting bootproto and startmode") do
+        let(:interface) { { "eth0" => { "bootproto" => "dhcp6", "startmode" => "onboot" } } }
+
+        it "sets both parameters correctly" do
+          expect(Yast::NetworkInterfaces).to receive(:Current=).with(
+            "BOOTPROTO" => "dhcp6",
+            "STARTMODE" => "onboot"
+          )
+          stub_write_interfaces
+          subject
+        end
+      end
+
+      context("setting IP address and netmask") do
+        let(:interface) { { "eth0" => { "ipaddr" => "TheIP/#{netmask}" } } }
+
+        context("with valid netmask in dot notation") do
```

Comment:
I'd like to see also a test for invalid input.

May be I'm to demanding, but network module is in bad shape and you'll see that sometimes it crashes even on obviously invalid user's input.